pkgs/development/ocaml-modules/bolt/default.nix

```diff
@@ -18,7 +18,8 @@ stdenv.mkDerivation rec {
     sha256 = "1c807wrpxra9sbb34lajhimwra28ldxv04m570567lh2b04n38zy";
   };
 
-  buildInputs = [ ocaml findlib ocamlbuild which camlp4 ];
+  nativeBuildInputs = [ ocaml findlib ocamlbuild which ];
+  buildInputs = [ camlp4 ];
```

Comment:
I suspect this belongs to nativeBuildInput, as it's a preprocessor.

Reply:
Me and @anmonteiro discussed this and we believe `camlp4` will be built and ran while building like ppxes. I would love to be proved wrong and be able to move more things to `nativeBuildInputs`

Reply:
Now that strictDeps is set, it only builds with camlp4 in nativeBuildInputs. That answers the question :)